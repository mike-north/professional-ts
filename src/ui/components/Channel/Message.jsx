import * as React from 'react';
import { formatTimestamp } from '../../../utils/date';

const Message = ({ user, date, body }) => (
  <div
    className="flex items-start px-6 py-2 text-sm hover-target hover:bg-gray-100 message"
    role="listitem"
  >
    <figure className="w-10 h-10 rounded overflow-hidden mr-3">
      <img
        className="message__user-avatar"
        src={user.iconUrl}
        alt={user.name}
      />
    </figure>

    <div className="flex-1">
      <h5 className="text-sm">
        <a
          href="#"
          className="message__user-name text-black font-bold no-underline hover:underline"
        >
          {user.name}
        </a>
        <span className="sr-only">at</span>
        <time className="message__timestamp text-gray-500 text-xs font-normal ml-1">
          {formatTimestamp(date)}
        </time>
      </h5>

      <p className="message__body text-black leading-normal">
        {body}
      </p>
    </div>

    <button
      className="message__delete-button border-transparent hover:border-red-400 show-on-hover hover:bg-red-100 border-1 rounded mb-1 pl-3 pr-2 py-1"
      aria-label="permanently delete this message"
    >
      ðŸ—‘
    </button>
  </div>
);

export default Message;
