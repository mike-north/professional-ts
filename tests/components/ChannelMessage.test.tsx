import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Message from '../../src/ui/components/Channel/Message';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Message
      user={{ name: 'Mike', iconUrl: '' }}
      date={new Date('01-01-2001')}
      body="Hello world!"
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
