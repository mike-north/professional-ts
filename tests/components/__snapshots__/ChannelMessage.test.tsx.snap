// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="flex items-start px-6 py-2 text-sm hover-target hover:bg-gray-100 message"
  role="listitem"
>
  <figure
    className="w-10 h-10 rounded overflow-hidden mr-3"
  >
    <img
      alt="Mike"
      className="message__user-avatar"
      src=""
    />
  </figure>
  <div
    className="flex-1"
  >
    <h5
      className="text-sm"
    >
      <a
        className="message__user-name text-black font-bold no-underline hover:underline"
        href="#"
      >
        Mike
      </a>
      <span
        className="sr-only"
      >
        at
      </span>
      <time
        className="message__timestamp text-gray-500 text-xs font-normal ml-1"
      >
        Jan 01, 2001 00:01:00 AM
      </time>
    </h5>
    <p
      className="message__body text-black leading-normal"
    >
      Hello world!
    </p>
  </div>
  <button
    aria-label="permanently delete this message"
    className="message__delete-button border-transparent hover:border-red-400 show-on-hover hover:bg-red-100 border-1 rounded mb-1 pl-3 pr-2 py-1"
  >
    ðŸ—‘
  </button>
</div>
`;
